File: easy school/teachers/teacher view.vb
Public Class teacher_view
    Dim sql As String = "SELECT  `name`, `national_id`,  `tel`, `email`,   `qualification`, `institution`, `year_out` FROM `teachers`"
    Private Sub Panel2_Paint(sender As Object, e As PaintEventArgs) Handles Panel2.Paint

    End Sub
    Sub fillter()
        Dim data As New studentsdatabase

        Dim fil As String = " where `name` regexp  '" & TextBox2.Text & "' or `national_id` regexp '" & TextBox2.Text & "'"
        Data.select_dgview(sql + fil, DataGridView1)
    End Sub

    Private Sub teacher_view_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim data As New studentsdatabase

        Data.select_dgview(sql, DataGridView1)
    End Sub

    Private Sub TextBox1_KeyPress(sender As Object, e As KeyPressEventArgs)

    End Sub


    Private Sub Button6_Click(sender As Object, e As EventArgs) Handles Button6.Click
        Dim data As New studentsdatabase

        Data.select_dgview(sql, DataGridView1)
        TextBox2.Text = ""
    End Sub

    Private Sub Button7_Click(sender As Object, e As EventArgs) Handles Button7.Click
        Me.Close()
    End Sub

    Private Sub Button3_Click_1(sender As Object, e As EventArgs) Handles Button3.Click
        Dim i As Integer
        Dim h As String
        If DataGridView1.RowCount > 0 Then
            i = DataGridView1.CurrentRow.Index
            With DataGridView1
                h = .Item(1, i).Value
            End With
            Try
                Dim data As New studentsdatabase
                Dim red As DataTable
                red = data.executeSQL("SELECT * FROM `teachers` WHERE `national_id`='" & h & "'")
                Dim tt As New teacher
                tt.selected(red)
                tt.Button3.Enabled = True
                tt.Button2.Enabled = False
                tt.ShowDialog()
            Catch ex As Exception

            End Try
        End If
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        Dim i As Integer
        Dim h As String
        If DataGridView1.RowCount > 0 Then
            i = DataGridView1.CurrentRow.Index
            With DataGridView1
                h = .Item(1, i).Value
                If MsgBox("are you sure you want to delete the record", MsgBoxStyle.YesNo, "confirm") = MsgBoxResult.Yes Then
                    Try
                        Dim data As New studentsdatabase
                        Dim red As DataTable
                        red = data.executeSQL("DELETE FROM `teachers` WHERE  `national_id`='" & h & "'")
                        Button6.PerformClick()
                        Dim tt As New teacher
                    Catch ex As Exception
                    End Try
                End If
            End With
        End If

    End Sub

    Private Sub TextBox2_TextChanged(sender As Object, e As EventArgs) Handles TextBox2.TextChanged
        If TextBox2.TextLength > 0 Then
            fillter()
        Else
            Button6.PerformClick()
        End If

    End Sub
End Class

File: easy school.ConvertedToC#/teachers/teacher view.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
namespace easy_school
{
	public partial class teacher_view
	{
		string sql = "SELECT  `name`, `national_id`,  `tel`, `email`,   `qualification`, `institution`, `year_out` FROM `teachers`";

		private void Panel2_Paint(object sender, PaintEventArgs e)
		{
		}
		public void fillter()
		{
			studentsdatabase data = new studentsdatabase();

			string fil = " where `name` regexp  '" + TextBox2.Text + "' or `national_id` regexp '" + TextBox2.Text + "'";
			data.select_dgview(sql + fil, DataGridView1);
		}

		private void teacher_view_Load(object sender, EventArgs e)
		{
			studentsdatabase data = new studentsdatabase();

			data.select_dgview(sql, DataGridView1);
		}


		private void TextBox1_KeyPress(object sender, KeyPressEventArgs e)
		{
		}


		private void Button6_Click(object sender, EventArgs e)
		{
			studentsdatabase data = new studentsdatabase();

			data.select_dgview(sql, DataGridView1);
			TextBox2.Text = "";
		}

		private void Button7_Click(object sender, EventArgs e)
		{
			this.Close();
		}

		private void Button3_Click_1(object sender, EventArgs e)
		{
			int i = 0;
			string h = null;
			if (DataGridView1.RowCount > 0) {
				i = DataGridView1.CurrentRow.Index;
				var _with1 = DataGridView1;
				h = _with1.Item(1, i).Value;
				try {
					studentsdatabase data = new studentsdatabase();
					DataTable red = null;
					red = data.executeSQL("SELECT * FROM `teachers` WHERE `national_id`='" + h + "'");
					teacher tt = new teacher();
					tt.selected(red);
					tt.Button3.Enabled = true;
					tt.Button2.Enabled = false;
					tt.ShowDialog();

				} catch (Exception ex) {
				}
			}
		}

		private void Button4_Click(object sender, EventArgs e)
		{
			int i = 0;
			string h = null;
			if (DataGridView1.RowCount > 0) {
				i = DataGridView1.CurrentRow.Index;
				var _with2 = DataGridView1;
				h = _with2.Item(1, i).Value;
				if (Interaction.MsgBox("are you sure you want to delete the record", MsgBoxStyle.YesNo, "confirm") == MsgBoxResult.Yes) {
					try {
						studentsdatabase data = new studentsdatabase();
						DataTable red = null;
						red = data.executeSQL("DELETE FROM `teachers` WHERE  `national_id`='" + h + "'");
						Button6.PerformClick();
						teacher tt = new teacher();
					} catch (Exception ex) {
					}
				}
			}

		}

		private void TextBox2_TextChanged(object sender, EventArgs e)
		{
			if (TextBox2.TextLength > 0) {
				fillter();
			} else {
				Button6.PerformClick();
			}

		}
		public teacher_view()
		{
			Load += teacher_view_Load;
			InitializeComponent();
		}
	}
}
